homeassistant/components/satel_integra/switch.py

```diff
@@ -57,58 +56,57 @@ class SatelIntegraSwitch(SwitchEntity):
     """Representation of an Satel switch."""
 
     _attr_should_poll = False
-
-    def __init__(self, controller, device_number, device_name, code, config_entry_id):
-        """Initialize the binary_sensor."""
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        controller: AsyncSatel,
+        device_number: int,
+        device_name: str,
+        code: str | None,
+        config_entry_id: str,
+    ) -> None:
+        """Initialize the switch."""
         self._device_number = device_number
         self._attr_unique_id = f"{config_entry_id}_switch_{device_number}"
-        self._name = device_name
         self._state = False
         self._code = code
         self._satel = controller
 
+        self._attr_name = device_name
+
     async def async_added_to_hass(self) -> None:
         """Register callbacks."""
-        async_dispatcher_connect(
-            self.hass, SIGNAL_OUTPUTS_UPDATED, self._devices_updated
+        self._state = self._device_number in self._satel.violated_outputs
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass, SIGNAL_OUTPUTS_UPDATED, self._devices_updated
+            )
         )
 
     @callback
-    def _devices_updated(self, zones):
+    def _devices_updated(self, outputs: dict[int, int]) -> None:
         """Update switch state, if needed."""
-        _LOGGER.debug("Update switch name: %s zones: %s", self._name, zones)
-        if self._device_number in zones:
-            new_state = self._read_state()
-            _LOGGER.debug("New state: %s", new_state)
+        if self._device_number in outputs:
+            new_state = outputs[self._device_number] == 1
             if new_state != self._state:
                 self._state = new_state
                 self.async_write_ha_state()
 
     async def async_turn_on(self, **kwargs: Any) -> None:
         """Turn the device on."""
-        _LOGGER.debug("Switch: %s status: %s, turning on", self._name, self._state)
         await self._satel.set_output(self._code, self._device_number, True)
+        self._state = True
         self.async_write_ha_state()
 
     async def async_turn_off(self, **kwargs: Any) -> None:
         """Turn the device off."""
-        _LOGGER.debug(
-            "Switch name: %s status: %s, turning off", self._name, self._state
-        )
         await self._satel.set_output(self._code, self._device_number, False)
+        self._state = False
         self.async_write_ha_state()
 
     @property
-    def is_on(self):
+    def is_on(self) -> bool | None:
         """Return true if device is on."""
-        self._state = self._read_state()
         return self._state
```

Comment:
Instead, replace `self._state` with `self._attr_is_on`, then you can remove this whole property :)